### PR TITLE
[skia] add svg fuzzer

### PR DESCRIPTION
**Pull request by werdnaH**
@kjlubick @jonathanmetzman

### CODE REVIEW

**Comment by kjlubick** [comment]
lgtm

File: projects/skia/svg_dom.options [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[libfuzzer]
+max_len = 30000
```
Comment (inferno-chromium) [inline projects/skia/svg_dom.options:2]: This needs to be checked in the fuzzer as otherwise wont work for other fuzzing engines.
if (size > 30000)
  return 0;

File: projects/skia/build.sh [hunk 2]
```diff
@@ -38,7 +38,7 @@ else
   CFLAGS= CXXFLAGS="-stdlib=libc++" cmake .. -D$CMAKE_SANITIZER=1
 fi
 
-make -j2 libGLESv2 libEGL
+make -j14 libGLESv2 libEGL
```
Comment (inferno-chromium) [inline projects/skia/build.sh:41]: Why this change.

File: projects/skia/BUILD.gn.diff [hunk 3]
```diff
@@ -451,4 +451,17 @@ test_app("skdescriptor_deserialize") {
...
+    "fuzz/Fuzz.cpp",
+    "fuzz/FuzzCommon.cpp",
+    "fuzz/oss_fuzz/FuzzSVG.cpp",
+  ]
+  deps = [
+    ":experimental_svg_model",
+    ":flags",
+    ":skia",
+  ]
 }
```
Comment (inferno-chromium) [inline projects/skia/BUILD.gn.diff:467]: new line.

File: projects/skia/build.sh (hunk 2 above)
Comment (kjlubick) [reply projects/skia/build.sh:41]: I suggested this change to make the compilations work faster locally for Zepeng. Do you think this will cause a problem in other places?

Comment (inferno-chromium) [reply projects/skia/build.sh:41]: ok fine then! i think it should be ok on builder.

**Comment by kjlubick** [comment]
The max_len thing I had been heretofore unaware of, so @werdnaH can probably fix *all* instances of that with one commit on the main Skia repo.

**Comment by inferno-chromium** [comment]
> The max_len thing I had been heretofore unaware of, so @werdnaH can probably fix _all_ instances of that with one commit on the main Skia repo.

Sounds good, then just remove the .options file from this CL, and fix in skia repo directly.
Also, just fyi, we are fixing CI so that it actually succeeds (travis already used to time out for large projects like skia).

**Comment by werdnaH** [comment]
I have fixed the issues, please take another look.

File: projects/skia/Dockerfile [hunk 4]
```diff
@@ -111,6 +113,8 @@ COPY sksl.options $SRC/skia/sksl2spirv.options
 COPY sksl.options $SRC/skia/sksl2metal.options
 COPY sksl.options $SRC/skia/sksl2pipeline.options
 
+COPY svg_dom.options $SRC/skia/svg_dom.options
```
Comment (kjlubick) [inline projects/skia/Dockerfile:116]: This will break

File: projects/skia/build.sh [hunk 5]
```diff
@@ -237,4 +238,8 @@ cp ./api_skdescriptor_seed_corpus.zip $OUT/api_skdescriptor_seed_corpus.zip
 # Don't have any examples of an SkDescriptor atm, so some random bytes is all we have.
 cp out/Fuzz/skdescriptor_deserialize $OUT/skdescriptor_deserialize
 cp ./skdescriptor_deserialize.options $OUT/skdescriptor_deserialize.options
-cp ./api_skdescriptor_seed_corpus.zip $OUT/skdescriptor_deserialize_seed_corpus.zip
\ No newline at end of file
+cp ./api_skdescriptor_seed_corpus.zip $OUT/skdescriptor_deserialize_seed_corpus.zip
+
+cp out/Fuzz/svg_dom $OUT/svg_dom
+cp ./svg_dom.options $OUT/svg_dom.options
```
Comment (kjlubick) [inline projects/skia/build.sh:244]: This will break

File: projects/skia/build.sh (hunk 2 above)
Comment (bungeman) [reply projects/skia/build.sh:41]: I'm trying to replace this with something that will intelligently scale at https://github.com/google/oss-fuzz/pull/3979 .
